While not strictly necessary because of patch-git_86dd2427, this patch helps
CMake determine the maximum size of the command line and determine whether
ninja generator needs to use .rsp files.

commit 5ec17e610609b8ca80cf8ce4f33f8c03ea48a090
Author: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date:   Fri Jul 15 12:46:53 2016 +0200

    Ninja: Generalize check for sysconf(3) call

    Use `sysconf(_SC_ARG_MAX)` wherever `_SC_ARG_MAX` is available instead
    of hard-coding an incomplete list of operating systems that implement
    it.  In particular, it is available on BSD platforms that were not
    previously listed.

--- Source/cmNinjaNormalTargetGenerator.cxx
+++ Source/cmNinjaNormalTargetGenerator.cxx
@@ -374,7 +374,7 @@ static int calculateCommandLineLengthLimit(int linkRuleLength)
 #ifdef _WIN32
     8000,
 #endif
-#if defined(__APPLE__) || defined(__HAIKU__) || defined(__linux)
+#if defined(_SC_ARG_MAX)
     // for instance ARG_MAX is 2096152 on Ubuntu or 262144 on Mac
     ((int)sysconf(_SC_ARG_MAX)) - 1000,
 #endif
